15 improve transaction doc (#23)
CARA-ch/emed-service-guide: docs/transactions/iti41.md

### Referencing external documents
When publishing CH-EMED-EPR documents, all references (from the key elements) shall be known to the eMedication service. All non-key elements are ignored by the eMedication service. 

This implies that the metadata cannot contain [Reference ids](https://profiles.ihe.net/ITI/TF/Volume3/ch-4.2.html#4.2.3.2.28). This option is not supported, and submission of a reference to an existing document is forbidden.

### Rules for each type of document
#### MTP
* Treatment duration (defined in [dosage](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-dosage.html)) should be within the period of validity of the document (defined by [`DocumentEntry.serviceStatTime`](https://profiles.ihe.net/ITI/TF/Volume3/ch-4.2.html#4.2.3.2.19) and [`DocumentEntry.serviceStopTime`](https://profiles.ihe.net/ITI/TF/Volume3/ch-4.2.html#4.2.3.2.20) metadata).
#### PRE

> * Effective time of [`documentationOf`](#FIXME) and [`serviceEvent`](#FIXME) has to be aligned with [`DocumentEntry.serviceStatTime`](https://profiles.ihe.net/ITI/TF/Volume3/ch-4.2.html#4.2.3.2.19) and [`DocumentEntry.serviceStopTime`](https://profiles.ihe.net/ITI/TF/Volume3/ch-4.2.html#4.2.3.2.20).

Author: I wasn't sure where to find the references to some of the document's attributes in the implementation guide. So I marked all of them with a #FIXME tag

Reviewer: `documentationOf ` and `serviceEvent `are CDA, just remove them

|`serviceStartTime`|O|Given as an [HL7 DTM](http://www.hl7.eu/refactored/dtDTM.html) in UTC. Shall be empty for APPC|
|`serviceStopTime`|O|Given as an [HL7 DTM](http://www.hl7.eu/refactored/dtDTM.html) in UTC. Shall be empty for APPC|
|`size`|O|-|
|`sourcePatientId`|R|Any id known or unknown to the local MPI, EPR-SPID and SSN are forbidden.|
|`sourcePatientInfo`|O|Ignored by the service.|
|`title`|R|-|
|`typeCode`|R|See [section](#ClassCode-metadata-to-use-for-each-document-type) below.|
|`uniqueId`|R|Must be globally unique and be a UUID. Must match the document id in the case of CH-EMED-EPR documents.|
|`URI`|O|-|
|`version`|O|If present shall be empty, the value is ignored and set by the document registry.|
|`DocumentEntry.patientId`|R|Must match `SubmissionSet.patientId` and be a XAD-PID ([see above](#SubmissionSet-metadata)).|
|`DocumentEntry.creationTime`|R|Must match document content creation time ([see below](#Metadata-values-check)).|
|`DocumentEntry.author`|R|Must be aligned with document content author ([see below](#Metadata-values-check)).|

> 

Reviewer: patientId, creationTime and author are repeated

## Publishing a CH-EMED-EPR document
This section details the specific rules to follow when publishing [CH-EMED-EPR documents](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/).

>  Only [MTP](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/document_mtp.html), [PRE](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/document_pre.html), [DIS](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/document_dis.html) and [PADV](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/document_padv.html) docs can be published. [PML](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/document_pml.html) and [PMLC](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/document_pmlc.html) can only be retrieved from the service. As a result, `Iti41QueryConverter.process->EmedFhirSpec.getDocumentType` returns `null` for `PML` and `PMLC` docs.

Reviewer: ```suggestion
 Only [MTP](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/document_mtp.html), [PRE](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/document_pre.html), [DIS](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/document_dis.html) and [PADV](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/document_padv.html) docs can be published. [PML](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/document_pml.html) and [PMLC](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/document_pmlc.html) can only be retrieved from the service.
```

### Referencing external documents
When publishing CH-EMED-EPR documents, all references (from the key elements) shall be known to the eMedication service. All non-key elements are ignored by the eMedication service. 

This implies that the metadata cannot contain [Reference ids](https://profiles.ihe.net/ITI/TF/Volume3/ch-4.2.html#4.2.3.2.28). This option is not supported, and submission of a reference to an existing document is forbidden.

### Creation times
[Creation time](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-composition-medicationtreatmentplan-definitions.html#Composition.date) of a document must be equal or greater than the creation time of the last document of the treatment chain, ie. the last published document targeting the treatment (either the original MTP if no other document, or the last document published referencing this treatment).
* Creation time of the document has to be in the past.
* All references to other documents or items have to refer to existing approved non deleted documents.

### Rules for each type of document
#### MTP

> * Treatment duration (defined in [dosage](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-dosage.html)) should be within the period of validity of the document, defined by [`DocumentEntry.serviceStatTime`](https://profiles.ihe.net/ITI/TF/Volume3/ch-4.2.html#4.2.3.2.19) and [`DocumentEntry.serviceStopTime`](https://profiles.ihe.net/ITI/TF/Volume3/ch-4.2.html#4.2.3.2.20) metadata

Reviewer: ```suggestion
* Treatment duration (defined in [dosage](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-dosage.html)) should be within the period of validity of the document, defined by [`DocumentEntry.serviceStartTime`](https://profiles.ihe.net/ITI/TF/Volume3/ch-4.2.html#4.2.3.2.19) and [`DocumentEntry.serviceStopTime`](https://profiles.ihe.net/ITI/TF/Volume3/ch-4.2.html#4.2.3.2.20) metadata
```

> * Treatment duration (defined in [dosage](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-dosage.html)) should be within the period of validity of the document, defined by [`DocumentEntry.serviceStartTime`](https://profiles.ihe.net/ITI/TF/Volume3/ch-4.2.html#4.2.3.2.19) and [`DocumentEntry.serviceStopTime`](https://profiles.ihe.net/ITI/TF/Volume3/ch-4.2.html#4.2.3.2.20) metadata

Reviewer: This check is not implemented and I wonder if it would make much sense (for this and any other document type).

#### PRE

> * Treatment duration (defined in [dosage](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-dosage-medicationrequest.html) - `CHEMEDEPRDosageMedicationRequest.repeat.bounds`) should be within the [period of validity of the document](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-medicationrequest.html) (`CHEMEDEPRMedicationRequest.validityPeriod`).

Reviewer: Now this check does not make any sense at all. Sorry if I provided it with the list, this is clearly wrong information on whatever document I pulled this from (or I just needed a coffee at the time). The `validityPeriod` of a prescription item refers to the validity of the prescription itself and not to the validity of the treatment in general, this is provided indeed by the `boundsPeriod` with the dosage which by the way, it can be altered by the prescription item, but it is advised against and will result in a warning (i.e. normally a prescription item will just _copy_ the dosage instructions provided by the treatment plan).

* Each PRE item should [refer to an MTP item](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-document-medicationprescription.html) (`CHEMEDEPRMedicationRequest.treatmentplan`) that must:
    * Exist (has already been published, and never deleted)
    * Be [approved](https://profiles.ihe.net/ITI/TF/Volume3/ch-4.2.html#4.2.3.2.2), ie. `DocumentEntry.availabilityStatus = approved`, 

>     * Be valid, ie. the submission date must be within the [MTP `dosage`'s](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-dosage.html) `boundsPeriod` (`CHEMEDEPRDosage.repeat.bounds`).

Reviewer: This check is not performed and I don't think it makes much sense, since MTP documents might be defining treatments in the future and hence the submission date would not be within the `boundsPeriod` if defined.

        * If the `boundsPeriod` is not defined, the entry is considered to be always active.
        * If only a [`startDate`](http://hl7.org/fhir/R4/datatypes-definitions.html#Period.start) is specified, the MTP is considered to be active if the current date is after that date.
    * Active, ie. `CHEMEDEPRDocumentMedicationTreatmentPlan.CHEMEDEPRMedicationStatement.status = active`
    * For the same patient.
 

> * [PRE document's date of validity](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-medicationrequest.html) (`CHEMEDEPRMedicationRequest.validityPeriod` ) must be after the [PRE document's creation time](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-composition-medicationprescription.html) (`CHEMEDCompositionMedicationPrescription.date`) and after the [start date of duration](https://build.fhir.org/ig/hl7ch/ch-emed//StructureDefinition-ch-emed-medicationstatement.html) (`CHEMEDMedicationStatement.effective[x].effectiveDateTime`) of treatment defined in the [MTP referenced by the PRE](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-document-medicationprescription.html) (`CHEMEDEPRMedicationRequest.treatmentplan`).

Reviewer: The check of the `validityPeriod` against the document creation time is not performed either. This one might make sense but it would have to be discussed to see whether it would be useful at all since in any case document submission should not be blocked by this and it would result in a warning, tops.

The check of the `validityPeriod` against the `effective` field is not implemented and it will never be since it is not supported by our IG and instead the treatment period is defined in the `boundsPeriod` of the base dosage instructions. A check of `validityPeriod` against the treatment period is performed and I don't think it makes much sense to perform such a check (arguably a prescription could be valid even before starting a treatment if the treatment is in the future?).

> * Code of [medication](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-medication.html) should match the code of medication of the [MTP referenced by the PRE](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-document-medicationprescription.html) (`CHEMEDEPRMedicationRequest.treatmentplan`).      

Author: @dvribeira I'm not sure about substitution here : does it not occur at DIS level ? Shouldn't PRE always be 100% aligned with MTP ?

> * If [lot number](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-medication.html) (`CHEMEDEPRMedication.batch`) is specified in referenced MTP, [the one in the PRE](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-medicationrequest.html) (`CHEMEDEPRMedicationRequest.medication[x]:medicationReference.batch`) should match it.

Author: @dvribeira : Same question as above about substitution

#### PADV
##### PADV against a provisional PRE
A "provisional PRE" is a prescription for which the [`CHEMEDEPRMedicationRequest.status`](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-medicationrequest.html) is `SUBMITTED` or `PROVISIONAL` and for which a PADV `OK` has not been completed yet. 

> * Medication cannot be changed.

Reviewer: not implemented and probably to be discussed

> * Dosage instructions cannot be changed.

Reviewer: same as previous

> * Substitution cannot be disallowed if the existing dispense includes one.

Reviewer: not implemented

* Observation code `REFUSE` cannot be issued.
#####  PADV `CANCEL`
* Not allowed against DIS or other PADV documents.

> 

Reviewer: Other implemented checks:

- PADV OK (sets targeted MTP or PRE status to ACTIVE):
  - against MTP: only allowed for treatment plans with status ACTIVE or SUSPENDED.
  - against PRE: only allowed for prescriptions with status SUBMITTED or PROVISIONAL
  - not allowed to target anything else than MTP or PRE
- PADV CHANGE (changes the treatment plan or prescription):
  - against MTP: 
    - the treatment status must be ACTIVE or SUSPENDED
    - the treatment must not have been prescribed
    - additionally, the treatment's status is set to ACTIVE as a result of the aggregation
  - against PRE:
    - the prescription's status must not be CANCELED or REFUSED
    - if the prescription's status is SUBMITTED, the aggregation sets it to ACTIVE as a result
  - not allowed to target anything else than MTP or PRE
- PADV CANCEL (sets status to CANCELED):
  - against MTP: 
    - sets also related prescriptions' status to CANCELED if their status was not already REFUSED or CANCELED.
    - treatment plan status must be ACTIVE or SUSPENDED
  - against PRE: not allowed for prescriptions whose status is already CANCELED or REFUSED. 
  - Not allowed to target anything else than MTP or PRE
- PADV REFUSE (sets status to REFUSED):
  - against MTP: 
    - sets also related prescriptions' status to REFUSED if the status was not already REFUSED or CANCELED.
    - treatment's status must be ACTIVE or SUSPENDED.
  - against PRE: the prescription's status must not be REFUSED, CANCELED or PROVISIONAL.
  - Not allowed to target anything else than MTP or PRE
- PADV SUSPEND (sets status to SUSPENDED): 
  - only allowed to target MTP entries
  - the treatment status must be ACTIVE
- PADV COMMENT: can target MTP, PRE and DIS entries only.

#### DIS

> * If [lot number](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-medication.html) (`batch`) is specified in referenced [MTP(`treatmentPlan`) or PRE (`prescription`)](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-medicationdispense.html), then it should match the one in the DIS.

Reviewer: same comment as for PRE check on this field

> ##### DIS against PRE or MTP:

Reviewer: actually, a DIS always targets an MTP and, optionally, a PRE

> * Medication cannot be different if substitution is not allowed by PRE. _This rule might be extended in the future_. 

Reviewer: This check is not performed for now. Will be implemented in the future and should result in a warning but not a fatal error.

> * Dispense cannot occur after the end of validity of the PRE document. _This rule might be extended in the future_.

Reviewer: Not performed for now. To be determined whether this should result in an error or a warning.

> * Dispense cannot occur after the end of the treatment if specified in the prescription item. _This rule might be extended in the future_.

Reviewer: Not performed for now. To be discussed and to be determined whether this should result in an error or a warning.

> * If referenced PRE is not provisional, it should be [dispensable](#FIXME).

Reviewer: I'm not sure what this means either. I guess this means that the referenced pre (if any) should have status PROVISIONAL or ACTIVE, but not sure. This would make sense but in any case it is not yet implemented and it should be further discussed.

* List of active [ingredients](https://build.fhir.org/ig/hl7ch/ch-emed//StructureDefinition-ch-emed-medication-medicationdispense-definitions.html#Medication.ingredient) should be the same than the one in the referenced [PRE](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-medicationdispense.html) (`prescription`). 

> 

Reviewer: The following check is also performed:

- If the treatment has been prescribed (at least one PRE document has been successfully submitted for it), then any DIS document targeting this treatment must reference one of its prescriptions.

### Roles
* [Patients](http://fhir.ch/ig/ch-epr-term/2.0.9/CodeSystem-2.16.756.5.30.1.127.3.10.6.html#2.16.756.5.30.1.127.3.10.6-PAT) and their [representatives](http://fhir.ch/ig/ch-epr-term/2.0.9/CodeSystem-2.16.756.5.30.1.127.3.10.6.html#2.16.756.5.30.1.127.3.10.6-REP) are allowed to publish only the following type of docs:
    * MTP
    * PADV with a `COMMENT` observation code with no restriction.
    * PADV with any other observation code only when referencing a document published by the same patient or a representative of the same patient.     
* [Healthcare professionals](http://fhir.ch/ig/ch-epr-term/2.0.9/CodeSystem-2.16.756.5.30.1.127.3.10.6.html#2.16.756.5.30.1.127.3.10.6-HCP) are allowed to publish only the following types of docs:
    * MTP
    * PRE
    * DIS
    * PADV		

### Metadata values check
Some values from the DocumentEntry are checked against the document (values in metadata and document must be equal):

| DocumentEntry | CH-EMED-EPR                                                  |
| ------------- | ------------------------------------------------------------ |
| author        | `Composition.author`                                         |

> | classCode     | `FIXME`                                           |

Author: for this one as well as formatCode and typeCode, your doc says : "The classCode, formatCode and typeCode attributes must match the document content." - I couldn't find the equivalent in the IG. Where does it occur in the documents?

Reviewer: Well it refers to the content in general, that means that if the class format and type are for an MTP document, then the FHIR content must be an MTP document bundle. Then the DocumentEntry.typeCode must match the Composition.type precisely.

	* representatives are not supported by the service yet.
* [Healthcare professionals](http://fhir.ch/ig/ch-epr-term/2.0.9/CodeSystem-2.16.756.5.30.1.127.3.10.6.html#2.16.756.5.30.1.127.3.10.6-HCP) (HCP) can only replace a document published by himself or by another HCP of same community of affiliation.
    * Current implementation: an HCP can replace a document published by any other HCP. Further rules and possible implementations are under discussion.
* [Document administrator](https://fhir.ch/ig/ch-epr-term/2.0.9/CodeSystem-2.16.756.5.30.1.127.3.10.6.html) may only replace a document published by an HCP of the same community of affiliation.
	* Not implemented in PMP, to be discussed, for now a document admin can replace any document.
* Replacing and replaced documents have to be of the same type (e.g. both MTP documents).
* Replacing and replaced documents have to refer to the same patient.
* Only documents at the end of a treatment chain can be replaced.
* Replaced document must exist in the repository.

> * A replacement document must can be linked only to elements of the same medication chain. In case a PRE document is replaced, it may be linked only to elements of the medication chains of the referenced MTP. 

Reviewer: ```suggestion
* A replacement document must be linked only to elements of the same medication chain. In case a PRE document is replaced, it may be linked only to elements of the medication chains of the referenced MTP. 
```

> * [Document administrator](https://fhir.ch/ig/ch-epr-term/2.0.9/CodeSystem-2.16.756.5.30.1.127.3.10.6.html) may only  can replace a document regardless of who published it.

Reviewer: ```suggestion
* [Document administrator](https://fhir.ch/ig/ch-epr-term/2.0.9/CodeSystem-2.16.756.5.30.1.127.3.10.6.html) can replace any document regardless of who published it.
```

## Publishing a PMP-APPC document
This section details the rules applicable to metadata when publishing [APPC documents](https://www.ihe.net/uploadedFiles/Documents/ITI/IHE_ITI_Suppl_APPC.pdf).

### Rules for APPC:
* Only  [Patients](http://fhir.ch/ig/ch-epr-term/2.0.9/CodeSystem-2.16.756.5.30.1.127.3.10.6.html#2.16.756.5.30.1.127.3.10.6-PAT) and their [representatives](http://fhir.ch/ig/ch-epr-term/2.0.9/CodeSystem-2.16.756.5.30.1.127.3.10.6.html#2.16.756.5.30.1.127.3.10.6-REP) or policy administrators can publish a new APPC document.
	* representatives are not supported by the service yet.
*  Only [Patients](http://fhir.ch/ig/ch-epr-term/2.0.9/CodeSystem-2.16.756.5.30.1.127.3.10.6.html#2.16.756.5.30.1.127.3.10.6-PAT) and their [representatives](http://fhir.ch/ig/ch-epr-term/2.0.9/CodeSystem-2.16.756.5.30.1.127.3.10.6.html#2.16.756.5.30.1.127.3.10.6-REP) can replace their own APPC documents (if it exists).
	* representatives are not supported by the service yet.
* Policy administrators of the reference community of the patient can replace any existing APPC document.

> * No APPC for the specified patient exists in the system, otherwise the document is refused.

Reviewer: Unless publishing a replacement.
Furthermore, as for CH-EMED-EPR, an APPC document to be replaced must:

- be approved (as before, this is always the case for documents in the PMP repository)
- not have `deletionStatus` = `deletionRequested`

> * APPC document structure described in specs §7.5

Reviewer: https://www.e-health-suisse.ch/fileadmin/user_upload/Dokumente/E/Implementation_Guide_eMedication_Architecture_Specs_013_Anhoerung.pdf

# ITI-41

The [ITI-41 (Provide and Register Document Set-b)](https://profiles.ihe.net/ITI/TF/Volume2/ITI-41.html) transaction allows a [content sender](https://profiles.ihe.net/ITI/TF/Volume2/ITI-41.html#3.41.2) to send documents to the eMedication service ([content receiver](https://profiles.ihe.net/ITI/TF/Volume2/ITI-41.html#3.41.2)). 

Generic [DocumentEntry metadata](https://profiles.ihe.net/ITI/TF/Volume3/ch-4.2.html#4.2.1.1) are described in the [documents section](documents.md). This page references additional constraints specific to ITI-41. See also IHE's documentation for [metadata in Document Sharing profiles](https://profiles.ihe.net/ITI/TF/Volume3/index.html).

## Publishing documents
### Document types
Two types of documents might be published to the service :

> - [CH-EMED-EPR](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/) documents. These documents contain the eMedication data. See also the [corresponding page](../emed/index.md) in this guide.

Reviewer: Only stable CH-EMED-EPR documents can be published, i.e. PML and PMLC documents are CH-EMED-EPR (on-demand) documents and cannot be provided by clients through ITI-41.

> - [APPC (Advanced Patient Privacy Consent)](https://www.ihe.net/uploadedFiles/Documents/ITI/IHE_ITI_Suppl_APPC.pdf) documents. These documents are used to let patients define who can see their data. See also the [corresponding page](../appc/index.md) in this guide.

Reviewer: not only see, since authorized HCPs will be able to submit documents for this patient, perhaps something like _define access rights to their data_ or any equivalent/better formula would be better suited

### SubmissionSet and relations
The eMedication service does not support folders and can only receive ONE document at a time. Hence, the [metadata](https://profiles.ihe.net/ITI/TF/Volume3/ch-4.1.html#4.1) must always contain a [SubmissionSet](https://profiles.ihe.net/ITI/TF/Volume3/ch-4.1.html#4.1.1.1) with a single [DocumentEntry](https://profiles.ihe.net/ITI/TF/Volume3/ch-4.1.html#4.1.1.3).

Only the following [associations](https://profiles.ihe.net/ITI/TF/Volume3/ch-4.1.html#4.1.2) are supported :
|Association Type|Usage|
|-------------------|-------|
|[urn:oasis:names:tc:ebxml-regrep:AssociationType:HasMember](https://profiles.ihe.net/ITI/TF/Volume3/ch-4.2.html#4.2.2.1.1)|[To link a DocumentEntry to the SubmissionSet](https://profiles.ihe.net/ITI/TF/Volume3/ch-4.1.html#4.1.2.1).|
|[urn:ihe:iti:2007:AssociationType:RPLC](https://profiles.ihe.net/ITI/TF/Volume3/ch-4.2.html#t4.2.2-1)|[To replace a previously published document](https://profiles.ihe.net/ITI/TF/Volume3/ch-4.1.html#4.1.2.2). Replaced document becomes deprecated.|

### Uniqueness

> Documents can be send to the eMedication service only once. Uniqueness is checked against [DocumentEntry's unique Id](https://profiles.ihe.net/ITI/TF/Volume3/ch-4.2.html#4.2.3.2.26), that must be an UUID.

Reviewer: ```suggestion
Documents can be sent to the eMedication service only once. Uniqueness is checked against [DocumentEntry's unique Id](https://profiles.ihe.net/ITI/TF/Volume3/ch-4.2.html#4.2.3.2.26), that must be an UUID.
```

> Documents can be sent to the eMedication service only once. Uniqueness is checked against [DocumentEntry's unique Id](https://profiles.ihe.net/ITI/TF/Volume3/ch-4.2.html#4.2.3.2.26), that must be an UUID.

Reviewer: `DocumentEntry.entryUUID` must also be unique

|`author.authorRole`|R|Possible values are defined in [ch-epr-term IG](http://fhir.ch/ig/ch-epr-term/2.0.9/ValueSet-DocumentEntry.authorRole.html). For APPC, only `PAT` (patient) and `REP` (Representative) are allowed.|
|`author.authorSpecialty`|O|Possible values are defined in [ch-epr-term IG](http://fhir.ch/ig/ch-epr-term/2.0.9/ValueSet-DocumentEntry.authorSpeciality.html).|
|`availabilityStatus`|O|The value set in the metadata is always ignored, and replaced by `approved`.|
|`comments`|O|-|
|`contentTypeCode`|R|Possible values are defined in [ch-epr-term IG](http://fhir.ch/ig/ch-epr-term/2.0.9/ValueSet-DocumentEntry.mimeType.html). Use `application/fhir+json` or `application/fhir+xml` for CH-EMED-EPR documents, and `text/xml` for APPC.|
|`entryUUID`|R|Must be globally unique.|
|`homeCommunityId`|O|Not used. Can be left blank or omitted.|
|`intendedRecipient`|O|Not used. Can be left blank or omitted.|
|`limitedMetadata`|O|Must be left bank or omitted.|
|`patientId`|R|Must be a XAD-PID.|
|`sourceId`|R|Globally unique and immutable OID identifier of the source.|
|`submissionTime`|R|[HL7 DTM](http://www.hl7.eu/refactored/dtDTM.html) value in UTC.|
|`title`|O|-|

> |`uniqueId`|R|Globally unique id.|

Reviewer: ```suggestion
```

|`mimeType`|O|Possible values are defined [ch-epr-term IG](http://fhir.ch/ig/ch-epr-term/2.0.9/ValueSet-DocumentEntry.mimeType.html). Use `application/fhir+json` or `application/fhir+xml` for CH-EMED-EPR documents, and `text/xml` for APPC.|
|`objectType`|R|Value is ignored and set to `Stable`.|
|`patientId`|R|Must match `SubmissionSet.patientId` and be a XAD-PID ([see above](#SubmissionSet-metadata)).|
|`practiceSettingCode`|R|See the [list of available codes](http://fhir.ch/ig/ch-epr-term/2.0.9/ValueSet-DocumentEntry.practiceSettingCode.html).|
|`referenceIdList`|O|Must be omitted or empty as referencing external documents is not permitted (see [Referencing external documents](#Referencing-external-documents) section).|
|`repositoryUniqueId`|O|If the attribute `repositoryUniqueId` is present and does not correspond to the `repositoryUniqueId` containing the documents the request is refused.|
|`serviceStartTime`|O|Given as an [HL7 DTM](http://www.hl7.eu/refactored/dtDTM.html) in UTC. Shall be empty for APPC|
|`serviceStopTime`|O|Given as an [HL7 DTM](http://www.hl7.eu/refactored/dtDTM.html) in UTC. Shall be empty for APPC|
|`size`|O|-|

> |`sourcePatientId`|R|Any id known or unknown to the local MP. EPR-SPID and SSN are forbidden.|

Reviewer: ```suggestion
|`sourcePatientId`|R|Any id known or unknown to the local MPI. EPR-SPID and SSN are forbidden.|
```

### Creation times
[Creation time](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-composition-medicationtreatmentplan-definitions.html#Composition.date) of a document must be equal or greater than the creation time of the last document of the treatment chain, ie. the last published document targeting the treatment (either the original MTP if no other document, or the last document published referencing this treatment).
* Creation time of the document has to be in the past.
* All references to other documents or items have to refer to existing approved non deleted documents.

### Rules for each type of document
#### PRE
* Each PRE item should [refer to an MTP item](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-document-medicationprescription.html) (`CHEMEDEPRMedicationRequest.treatmentplan`) that must:
    * Exist (has already been published, and never deleted)
    * Be [approved](https://profiles.ihe.net/ITI/TF/Volume3/ch-4.2.html#4.2.3.2.2), ie. `DocumentEntry.availabilityStatus = approved`, 

>     * Preferably  be valid, ie. the submission date should be within the [MTP `dosage`'s](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-dosage.html) `boundsPeriod` (`CHEMEDEPRDosage.repeat.bounds`).

Reviewer: ```suggestion
    * Preferably  be valid, i.e. the submission date should be within the [MTP `dosage`'s](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-dosage.html) `boundsPeriod` (`CHEMEDEPRDosage.repeat.bounds`).
```

>     * Preferably  be valid, i.e. the submission date should be within the [MTP `dosage`'s](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-dosage.html) `boundsPeriod` (`CHEMEDEPRDosage.repeat.bounds`).

Reviewer: ```suggestion
    * Preferably be valid, i.e. the submission date should be within the [MTP `dosage`'s](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-dosage.html) `boundsPeriod` (`CHEMEDEPRDosage.repeat.bounds`).
```

	* representatives are not supported by the service yet.
* [Healthcare professionals](http://fhir.ch/ig/ch-epr-term/2.0.9/CodeSystem-2.16.756.5.30.1.127.3.10.6.html#2.16.756.5.30.1.127.3.10.6-HCP) (HCP) can only replace a document published by himself or by another HCP of same community of affiliation.
    * Current implementation: an HCP can replace a document published by any other HCP. Further rules and possible implementations are under discussion.
* [Document administrator](https://fhir.ch/ig/ch-epr-term/2.0.9/CodeSystem-2.16.756.5.30.1.127.3.10.6.html) may only replace a document published by an HCP of the same community of affiliation.
	* Not implemented in PMP, to be discussed, for now a document admin can replace any document.
* Replacing and replaced documents have to be of the same type (e.g. both MTP documents).
* Replacing and replaced documents have to refer to the same patient.
* Only documents at the end of a treatment chain can be replaced.
* Replaced document must exist in the repository.

> * A replacement document must can be linked only to elements of the same medication chain. In case a PRE document is replaced, it may be linked only to elements of the medication chains of the referenced MTP. 

Reviewer: ```suggestion
* A replacement document must can be linked only to elements of the same medication chain. In case a PRE document is replaced, it may be linked only to elements of the medication chain of the referenced MTP. 
```

### Rules for each type of document
#### PRE
* Each PRE item should [refer to an MTP item](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-document-medicationprescription.html) (`CHEMEDEPRMedicationRequest.treatmentplan`) that must:
    * Exist (has already been published, and never deleted)
    * Be [approved](https://profiles.ihe.net/ITI/TF/Volume3/ch-4.2.html#4.2.3.2.2), ie. `DocumentEntry.availabilityStatus = approved`, 
    * Preferably  be valid, ie. the submission date should be within the [MTP `dosage`'s](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-dosage.html) `boundsPeriod` (`CHEMEDEPRDosage.repeat.bounds`).
        * If the `boundsPeriod` is not defined, the entry is considered to be always active.
        * If only a [`startDate`](http://hl7.org/fhir/R4/datatypes-definitions.html#Period.start) is specified, the MTP is considered to be active if the current date is after that date.
    * Active, ie. `CHEMEDEPRDocumentMedicationTreatmentPlan.CHEMEDEPRMedicationStatement.status = active`
    * For the same patient.

> * Code of [medication](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-medication.html) should match the code of medication of the [MTP referenced by the PRE](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-document-medicationprescription.html) (`CHEMEDEPRMedicationRequest.treatmentplan`). _This rule is not enforced and might be extended in the future_.    

Author: @dvribeira I'm not sure about substitution here : does it not occur at DIS level ? Shouldn't PRE always be 100% aligned with MTP ?

Reviewer: - Yes, substitution should not be done at the PRE level and in fact it cannot flagged. Having said that, what should we do if a prescription does not provide the same medication as the MTP?
  - Should we reject the PRE submission altogether?
  - Should we accept it with a warning and then:
    - Accept the medication of the PRE?
    - Ignore the medication of the PRE?
In my opinion the document should be accepted along with the specified medication but with a warning being returned upon submission. Having said that, this can be discussed with everybody and more particularly with professionals.

> * If [lot number](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-medication.html) (`CHEMEDEPRMedication.batch`) is specified in referenced MTP, [the one in the PRE](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-medicationrequest.html) (`CHEMEDEPRMedicationRequest.medication[x]:medicationReference.batch`) should match it.  _This rule is not enforced and might be extended in the future_.

Author: @dvribeira : Same question as above about substitution

Reviewer: In general lot number does not make much sense to be specified in MTP or PRE, though I guess this rule is here for some reason. However I don't think this is substitution, since it is just the lot number on the packaging, the medication is not changed, it is exactly the same product, even the packaging if specified (e.g. GTIN included).
So yes, I guess the question is here is why would MTP or PRE include a lot number, perhaps this is more a question for professionals than for me.

### Rules for each type of document
#### MTP
* Treatment duration (defined in [dosage](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-dosage.html)) should be within the period of validity of the document (defined by [`DocumentEntry.serviceStatTime`](https://profiles.ihe.net/ITI/TF/Volume3/ch-4.2.html#4.2.3.2.19) and [`DocumentEntry.serviceStopTime`](https://profiles.ihe.net/ITI/TF/Volume3/ch-4.2.html#4.2.3.2.20) metadata).
#### PRE
* Effective time of [`documentationOf`](#FIXME) and [`serviceEvent`](#FIXME) has to be aligned with [`DocumentEntry.serviceStatTime`](https://profiles.ihe.net/ITI/TF/Volume3/ch-4.2.html#4.2.3.2.19) and [`DocumentEntry.serviceStopTime`](https://profiles.ihe.net/ITI/TF/Volume3/ch-4.2.html#4.2.3.2.20).
* Treatment duration (defined in [dosage](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-dosage.html)) should be within the period of validity of the document (defined by [`DocumentEntry.serviceStatTime`](https://profiles.ihe.net/ITI/TF/Volume3/ch-4.2.html#4.2.3.2.19) and [`DocumentEntry.serviceStopTime`](https://profiles.ihe.net/ITI/TF/Volume3/ch-4.2.html#4.2.3.2.20) metadata).

> * Each PRE item should refer to an MTP item that must be a [available](#FIXME) [approved](#FIXME) [valid](#FIXME) [active](#FIXME) and for the same patient

Reviewer: - available -> maybe rephrase it to say that it exists in the PMP repository, it was previously published and its deletion has not been requested
- approved -> `DocumentEntry.availabilityStatus = approved` i.e. not deprecated/replaced
- valid -> `NOW BETWEEN CHEMEDEPRMedicationStatement.dosage:baseEntry.timing.repeat.boundsPeriod.start AND CHEMEDEPRMedicationStatement.dosage:baseEntry.timing.repeat.boundsPeriod.end` (start et end sont optionnelles, si on n'a pas timing/repeat/boundsPeriod donc l'entrée reste toujours active. Si on a start et pas d'end, donc l'entrée est active si NOW >= start. Si on a end e pas de start, donc l'entrée est active si NOW <= end.
- active -> `CHEMEDEPRMedicationStatement.status = active`

    * Not blocking issues should not prevent publication but should be notified to service administration to address the resulting ad-hoc issue    
* [Validity of document](#FIXME) has to be after the [date of the PRE document](#FIXME) and the [start date of duratio](#FIXME)n of treatment of parent MTP 

> * [Code of medication](#FIXME) should match the code of medication of referenced MTP.      

Reviewer: `CHEMEDEPRMedication.code`
I don't think this is checked at the moment, I should verify this.

> * If [lot number](#FIXME) is specified in referenced MTP, the PRE item mast match it.

Reviewer: `CHEMEDEPRMedication.batch.lotNumber`
Not sure if this check is done at the moment.

#### PADV
##### PADV against a [provisional PRE](#FIXME)
* Medication cannot be changed.
* Dosage instructions cannot be changed.
* Substitution cannot be disallowed if the existing dispense includes a substitution.
* Observation code `REFUSE` cannot be issued.
#####  PADV `CANCEL`
* Not allowed against DIS or other PADV documents.

#### DIS

> * Dispense [status](#FIXME) has to be coherent with previous DIS documents for the same prescription.

Reviewer: `CHEMEDEPRMedicationDispense.status`
Not sure what the actual rule here should be, probably to de defined by professionals. In any case these would be warnings in principle and would not prevent publication.
I would suggest to remove this one for now.

> * If [lot number](#FIXME) is specified in parent MTP or PRE, lot number must match.

Reviewer: Same as for https://github.com/CARA-ch/emed-service-guide/pull/23/commits/9963dbb21b43ea871752c3cba1bae61a5805b0da#r1315836318
This is not checked at the moment and would probably just result in a warning if implemented.

##### DIS against PRE:
* Medication cannot be different if substitution is not allowed by PRE. 

>     * Verification Is performed on the [GTIN](#FIXME) of the medication (not of the package). 

Reviewer: Not implemented yet. Maybe add a to-be-defined disclaimer, since the precise rules on these should be refined with professional input.

    * If a substitution other than `none` is specified, no validation is performed.

> * Dispense cannot occur after the [end of validity of the PRE](#FIXME) document.

Reviewer: maybe just add a TBD

> * Dispense cannot occur after the [end of the treatment](#FIXME) if specified in the [prescription item](#FIXME).

Reviewer: keep this on hold until discussion regarding authorship timestamps is sorted out

* If referenced PRE is not [provisional](#FIXME), it should be [dispensable](#FIXME).

> * List of active ingredients should be the same than the one in the [parent PRE](#FIXME).

Reviewer: Similar to the medication one... the list of ingredients is given in `CHEMEDEPRMedication.ingredient` of which `CHEMEDEPRMedication.ingredient.isActive` is true.

Reviewer: This is not checked atm. Should it be implemented, it will result in a warning.
Note this check should be done even if substitution is allowed.

> * If substitution occurs, the [medication code (GTIN)](#FIXME) should be the same than the medication code from the [parent PRE](#FIXME).

Reviewer: Not sure this is correct actually... it does not make much sense to me.

Reviewer: Just remove this for now.

##### Against MTP:
* Medication cannot be different if [substitution](#FIXME) is not allowed by the MTP item. Verification is done on medication's [GTIN](#FIXME).

> * Dispense cannot occur after the [end of the treatment](#FIXME) if specified in the MTP item.

Reviewer: as against PRE

* Referenced MTP must be [approved](#FIXME) and [active](#FIXME) and for the same patient.
* [List of active ingredients](#FIXME) should be the same than the one in the [parent MTP](#FIXME).

> * If substitution occurs, the [medication code (GTIN)](#FIXME) should be the same than the medication code from the [parent MTP](#FIXME).

Reviewer: as against PRE

### Roles
* [Patients](#FIXME) and their [representatives](#FIXME) are allowed to publish only the following type of docs:
    * MTP

>     * DIS

Reviewer: Actually, atm, patients and their representatives are not allowed to publish DIS docs either.

    * PADV with a COMMENT observation code with no restriction.
    * PADV with any other observation code only when referencing a document published by the same patient or a representative of the same patient.     
* [Healthcare providers](#FIXME) are allowed to publish only the following types of docs:
    * MTP
    * PRE
    * DIS
    * PADV		

> * [Creation time of the document](#FIXME) must be equal or greater than the creation time of the last document of the [treatment chain](#FIXME).

Reviewer: - Creation time of the document: `CHEMEDDocumentMedicationPrescription.Composition.date` and/or `CHEMEDDocumentMedicationPrescription.timestamp`.
- The last document of the treatment chain is the last published document targeting the treatment (either the original MTP if no other document, or the last document published referencing this treatment).

> The class and type codes are from the SNOMED CT system: `2.16.840.1.113883.6.96`.

Reviewer: the system for class and type codes is correct, but we should also state that the system for the `formatCode `is `2.16.756.5.30.1.127.3.10.10` for completion
